### PR TITLE
Fix reproducability issue of seeded StableDiffusion runs

### LINKED ISSUE #844: StableDiffusion() model fails to reproduce same image from same prompt and seed
The `StableDiffusion()` model fails to reproduce the same image when provided with the same `seed` value and `prompt`. 

This issue is likely linked to inconsistency with the `tf.random.normal()` function. This function appears in line 187 of `stable_diffusion.py` https://github.com/keras-team/keras-cv/blob/master/keras_cv/models/generative/stable_diffusion/stable_diffusion.py. 

I recommend either 1) repair `tf.random.normal()` to consistently provide the same output with the same `seed` value, or 2) replace that function with `tf.random.stateless_normal()` on line 187 of `stable_diffusion.py` for consistent outputs with a fixed `seed` value.

An example of the inconsistency of `tf.random.normal()` is shown in the snippet below:
```python
import numpy as np
import tensorflow as tf
​
np.sum(tf.random.normal((1,2,3,4), seed=123))
OUT: -4.219632
np.sum(tf.random.normal((1,2,3,4), seed=123))
OUT: 1.8580852
```

Conversely, `tf.random.stateless_normal()` is consistent when using the same `seed` value:
```python
np.sum(tf.random.stateless_normal((1,2,3,4), seed=[123,1]))
OUT: 2.5994601
np.sum(tf.random.stateless_normal((1,2,3,4), seed=[123,1]))
OUT: 2.5994601
```

### PR DESCRIPTION
**Pull request by ianstenbit**
Fixes #844 

Ran tests manually locally (had to adjust golden values as we no longer need to set TF seed for this)

### CODE REVIEW

File: keras_cv/models/generative/stable_diffusion/stable_diffusion.py [hunk 1]
```diff
@@ -327,9 +327,15 @@ def _get_initial_alphas(self, timesteps):
         return alphas, alphas_prev
 
     def _get_initial_diffusion_noise(self, batch_size, seed):
-        return tf.random.normal(
-            (batch_size, self.img_height // 8, self.img_width // 8, 4), seed=seed
-        )
+        if seed:
```
Comment (fchollet) [inline keras_cv/models/generative/stable_diffusion/stable_diffusion.py:330]: if seed is not None (it might be 0 which is a valid seed)

Comment (ianstenbit) [reply keras_cv/models/generative/stable_diffusion/stable_diffusion.py]: Agh thank you for catching this -- done

**Review by LukeWood — APPROVED** [review]

**Comment by LukeWood** [comment]
Looks good to me!

**Pull request merged**